internal/updater/bundle-deployment/updater.go

```diff
@@ -69,3 +69,10 @@ func EnsureInstalledName(bundleName string) UpdateStatusFunc {
 		return true
 	}
 }
+
+func EnsureIncreaseGeneration() UpdateStatusFunc {
+	return func(status *rukpakv1alpha1.BundleDeploymentStatus) bool {
+		status.ObservedGeneration++
+		return true
+	}
+}
```

Comment:
Let's make it look like this: https://github.com/operator-framework/rukpak/blob/aaba57539dc3cb75e7f26870ccd3af48ded66707/internal/updater/bundle/updater.go#L80-L88

There's not a guarantee that the controller sees every generation of the object. It's also possible that the same generation of an object will be reconciled multiple times. Therefore, a simple increment like this would result in an incorrect skew between generation and observedGeneration.

Reply:
I update.  Thanks!